src/v3.rs

```diff
@@ -201,14 +208,7 @@ impl Message {
 
     /// Add content to the message.
     pub fn add_content(mut self, c: Content) -> Message {
-        match self.content {
-            None => {
-                let mut content = Vec::new();
-                content.push(c);
-                self.content = Some(content);
-            }
-            Some(ref mut content) => content.push(c),
-        };
+        self.content.get_or_insert_with(Vec::new).push(c);
```

Comment:
This is a great change. This was one of my first Rust crates and it shows by not knowing how concise I could have made this function! :)